### PR TITLE
chore(napi/transform): support wasm build

### PR DESCRIPTION
**Pull request by Dunqing**
~~Encountered the same problem as https://github.com/oxc-project/oxc-resolver/issues/190. Fixed in napi@2.16.7, but oxc use `3.x`~~

Fixed in napi@3.0.0-alpha.3

### CODE REVIEW

**Comment by Dunqing** [comment]
* **#3933** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/3933?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Dunqing and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: .github/actions/pnpm/action.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+name: Pnpm
```
Comment (Boshen) [inline .github/actions/pnpm/action.yml:1]: This is not needed, can inline this.

**Review by Boshen — CHANGES_REQUESTED** [review]

File: .github/actions/pnpm/action.yml (hunk 1 above)
Comment (Boshen) [reply .github/actions/pnpm/action.yml:1]: Or maybe refactor the other scripts later?

Comment (Dunqing) [reply .github/actions/pnpm/action.yml:1]: Yes, refactor later

**Pull request merged**